### ISSUE TITLE
entr -d inhibited by rm

### ISSUE THREAD

**Issue by q3cpma** [eradman/entr#8]
Hello,

I'm trying to replace the use of inotifywait by entr in order to be more portable and I noticed a problem:
```
user@gentoo-zen2700x> touch 1
user@gentoo-zen2700x> echo . | entr -pdn echo HELLO &
[1] 13297 13298
user@gentoo-zen2700x> rm 1
user@gentoo-zen2700x> touch 2
user@gentoo-zen2700x> touch 3
user@gentoo-zen2700x> HELLO                                                                                                             ~/Programming/mus
entr: directory altered

[1]  + done       echo . |
       exit 2     entr -pdn echo HELLO
```
As you can see, entr doesn't react to `touch 2` for some reason. Also, for some reason, entr feels a lot slower than inotifywait -e CREATE to react. Any reason for that?

Tested with entr-4.2 on Gentoo amd64.

**Comment by q3cpma** [comment]
Also, a bit unrelated, but is there a way to exit when a file is modified, like `-d` does?

**Comment by q3cpma** [comment]
Well, I currently use `entr -ns 'kill $PPID'` but it feels like a hack.

**Comment by eradman** [comment]
`entr -s 'kill $PPID'` is what I would recommend if you want one-shot behavior.  I have thought about adding a one-shot flag, but also feel there is nothing very wrong with this workaround.

As for your first question: entr doesn't respond to `touch 2` because it exits after you run `rm 1`.  Instead of backgrounding the process does the following work?
 
    while sleep 0.1; do
      echo "starting"
      echo . | entr -pds 'kill $PPID'
    done

`entr` does reacts immediately to some events. If a file is created with the `-d` option entr will delay in for the directory contents to stabilize. This is so that an operation such as `git checkout <branch>` does trigger the execution of a command before all of the files have finished changing.

**Comment by eradman** [comment]
@q3cpma : I implemented a one-shot option (-z) in commit 3699800. For more details see issue #23

**Comment by q3cpma** [comment]
Thanks, this could be useful. Currently, I just poll because portability is the
goal, but I might conditionally use entr.
